src/app/components/RealtimeDetection.jsx

```diff
@@ -1,47 +1,71 @@
+"use client";
+
+import { useState } from "react";
+import { useQuery } from "@tanstack/react-query";
 import { Search } from "lucide-react";
-import { Card } from "./ui/card";
+import ThreatTable from "./ThreatTable";
+import ThreatModal from "./ThreatModal";
+
+const fetchThreats = async () => {
+  const res = await fetch("/api/threats", { cache: "no-store" });
+  if (!res.ok) throw new Error("Failed to fetch");
+  const data = await res.json();
+  return (data.alerts || []).map((t) => ({
+    type: t.type || "Unknown",
+    time: t.createdDateTime || new Date().toISOString(),
+    severity: t.severity || "low",
+    status: t.status || "Pending",
+    affected: t.affected || "Unknown endpoint",
+  }));
+};
+
+export default function RealtimeDetection() {
+  const [filter, setFilter] = useState("");
+  const [selectedThreat, setSelectedThreat] = useState(null);
+
+  const { data: threatLog = [], isLoading, error } = useQuery({
+    queryKey: ["threats"],
+    queryFn: fetchThreats,
+    staleTime: 10000,
```

Comment:
Do you really need to add staleTime? Would the default staleTime not work well for your usecase?